source/h2_connection.c

```diff
@@ -435,25 +408,9 @@ static void s_handler_destroy(struct aws_channel_handler *handler) {
         struct aws_h2_frame *frame = AWS_CONTAINER_OF(node, struct aws_h2_frame, node);
         aws_h2_frame_destroy(frame);
     }
-    while (!aws_linked_list_empty(&connection->thread_data.pending_settings_queue)) {
-        struct aws_linked_list_node *node = aws_linked_list_pop_front(&connection->thread_data.pending_settings_queue);
-        struct aws_h2_pending_settings *pending_settings = AWS_CONTAINER_OF(node, struct aws_h2_pending_settings, node);
-        /* fire the user callback with error */
-        if (pending_settings->on_completed) {
-            pending_settings->on_completed(
-                &connection->base, AWS_ERROR_HTTP_CONNECTION_CLOSED, pending_settings->user_data);
-        }
-        aws_mem_release(connection->base.alloc, pending_settings);
-    }
-    while (!aws_linked_list_empty(&connection->thread_data.pending_ping_queue)) {
-        struct aws_linked_list_node *node = aws_linked_list_pop_front(&connection->thread_data.pending_ping_queue);
-        struct aws_h2_pending_ping *pending_ping = AWS_CONTAINER_OF(node, struct aws_h2_pending_ping, node);
-        /* fire the user callback with error */
-        if (pending_ping->on_completed) {
-            pending_ping->on_completed(
-                &connection->base, 0 /*fake rtt*/, AWS_ERROR_HTTP_CONNECTION_CLOSED, pending_ping->user_data);
-        }
-        aws_mem_release(connection->base.alloc, pending_ping);
+    if (connection->thread_data.init_pending_settings) {
+        /* if initial settings is not ACKed by peer, we need to clear the memory here */
```

Comment:
trivial: /* if initial settings were never sent, we need to clear the memory here */